src/Stratis.Bitcoin.Features.SmartContracts/Rules/SmartContractCoinViewRuleLogic.cs

```diff
@@ -102,6 +126,14 @@ public async Task RunAsync(Func<RuleContext, Task> baseRunAsync, RuleContext con
         /// </summary>
         public void UpdateCoinView(Action<RuleContext, Transaction> baseUpdateUTXOSet, RuleContext context, Transaction transaction)
         {
+            // We already have results for this block. No need to do any processing other than updating the UTXO set.
+            if (this.cachedResults != null)
+            {
```

Comment:
This bypasses `ValidateRefunds`, `ValidateGeneratedTransaction` and `ValidateSubmittedTransaction`.

Reply:
Right, I'm willing to say that it's okay as we are trusting our own miner, it's part of the same program.